test/test_nans.h

```diff
@@ -11,8 +11,8 @@ template <typename T> struct MeanTest : public ::testing::Test {
    * For any integer input type we expect the return type to be double
    */
   using RetType = std::conditional_t<std::is_integral_v<T>, double, T>;
-  /// Held type supports nan testing. All FP types -> true
-  constexpr static bool TestNans = !std::numeric_limits<T>::is_integer;
+  /// Held type supports nan testing.
+  constexpr static bool TestNans = std::numeric_limits<T>::is_iec559;
```

Comment:
👍 , should we be more explicit `has_quiet_NaN`?

Reply:
That would match the actual tests better. I'll change it for this PR.

But `isnan` can also handle signaling NaN's. Would it make sense to expand the `nanmean` test o include that? While we're at it, we might also conditionally disable tests of `isxxx`...

Reply:
I think so.